src/cpu-utils.h

```diff
@@ -55,3 +68,42 @@ static inline int get_avail_nprocs()
         nproc = 1;
     return nproc;
 }
+
+/**
+ * @brief Get the thread number per physical processor.
+ *
+ * Acquire the thread number by doing the division to the logical processor number with physical processor number.
+ * @return The thread number per physical processor.
+ * @retval 1 Hyperthreading disabled.
+ * @retval 2 Hyperthreading enabled.
+ * @retval -1 Unexpected error.
+ */
+static inline int get_nthds_per_physic_proc()
+{
+    char sib[4], cores[4];
+    int siblings, cpu_cores;
+    FILE *fd;
+    fd = popen("cat /proc/cpuinfo | grep 'siblings' | uniq | awk '{printf $3}'",
```

Comment:
It is Linux specific. I would expect both macOS and GNU/Linux can facilitate.

Reply:
I'll change the command to `LC_ALL=C lscpu | grep 'CPU(s)' | awk '{printf $2}'` on GNU/Linux.